group-coordinator/src/test/java/org/apache/kafka/coordinator/group/streams/topics/CopartitionedTopicsEnforcerTest.java

```diff
@@ -0,0 +1,203 @@
+        final CopartitionedTopicsEnforcer validator = new CopartitionedTopicsEnforcer(LOG_CONTEXT, CopartitionedTopicsEnforcerTest::firstSecondTopicInconsistent);
+        assertThrows(StreamsInconsistentInternalTopicsException.class, () -> validator.enforce(Utils.mkSet("first", "second"),
+            Collections.emptyMap()));
+    }
+
+
+    @Test
+    public void shouldEnforceCopartitioningOnRepartitionTopics() {
+        final CopartitionedTopicsEnforcer validator = new CopartitionedTopicsEnforcer(LOG_CONTEXT, CopartitionedTopicsEnforcerTest::firstSecondTopicConsistent);
+        final ConfiguredInternalTopic config = createTopicConfig("repartitioned", 10);
+
+        validator.enforce(Utils.mkSet("first", "second", config.name()),
+            Collections.singletonMap(config.name(), config));
+
+        assertEquals(Optional.of(2), config.numberOfPartitions());
+    }
+
+
+    @Test
+    public void shouldSetNumPartitionsToMaximumPartitionsWhenAllTopicsAreRepartitionTopics() {
+        final CopartitionedTopicsEnforcer validator = new CopartitionedTopicsEnforcer(LOG_CONTEXT, CopartitionedTopicsEnforcerTest::emptyTopicPartitionProvider);
+        final ConfiguredInternalTopic one = createTopicConfig("one", 1);
+        final ConfiguredInternalTopic two = createTopicConfig("two", 15);
+        final ConfiguredInternalTopic three = createTopicConfig("three", 5);
+        final Map<String, ConfiguredInternalTopic> internalTopicConfigs = new HashMap<>();
+
+        internalTopicConfigs.put(one.name(), one);
+        internalTopicConfigs.put(two.name(), two);
+        internalTopicConfigs.put(three.name(), three);
+
+        validator.enforce(Utils.mkSet(one.name(),
+                two.name(),
+                three.name()),
+            internalTopicConfigs
+        );
+
+        assertEquals(Optional.of(15), one.numberOfPartitions());
+        assertEquals(Optional.of(15), two.numberOfPartitions());
+        assertEquals(Optional.of(15), three.numberOfPartitions());
+    }
+
+    @Test
+    public void shouldThrowAnExceptionIfInternalTopicConfigsWithEnforcedNumOfPartitionsHaveDifferentNumOfPartitions() {
+        final CopartitionedTopicsEnforcer validator = new CopartitionedTopicsEnforcer(LOG_CONTEXT, CopartitionedTopicsEnforcerTest::firstSecondTopicConsistent);
+        final ConfiguredInternalTopic topic1 = createInternalTopicConfigWithEnforcedNumberOfPartitions("repartitioned-1", 10);
+        final ConfiguredInternalTopic topic2 = createInternalTopicConfigWithEnforcedNumberOfPartitions("repartitioned-2", 5);
+
+        final StreamsInconsistentInternalTopicsException ex = assertThrows(
+            StreamsInconsistentInternalTopicsException.class,
+            () -> validator.enforce(Utils.mkSet(topic1.name(), topic2.name()),
+                Utils.mkMap(Utils.mkEntry(topic1.name(), topic1),
+                    Utils.mkEntry(topic2.name(), topic2)))
```

Comment:
nit for readability:
```suggestion
                Utils.mkMap(
                    Utils.mkEntry(topic1.name(), topic1),
                    Utils.mkEntry(topic2.name(), topic2)
                )
            )
```